File: z-clients/javascript/src/models/httpSinkConfig.ts
// this file is @generated
import {
    type HttpMethod,
    HttpMethodSerializer,
} from './httpMethod';
/**
* Configuration for an HTTP sink. The `url`, `headers`, and `body` are templates rendered
* per-message (see [`diom_core::template_str`]).
*/
export interface HttpSinkConfig {
    /** Destination URL. */
    url: string;
    method?: HttpMethod;
    headers?: { [key: string]: string };
    /** Templated request body. When absent, the raw message value bytes are sent unchanged. */
    body?: string | null;
}

export const HttpSinkConfigSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): HttpSinkConfig {
        return {
            url: object['url'],
            method: object['method'] != null ? HttpMethodSerializer._fromJsonObject(object['method']): undefined,
            headers: object['headers'],
            body: object['body'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: HttpSinkConfig): any {
        return {
            'url': self.url,
            'method': self.method != null ? HttpMethodSerializer._toJsonObject(self.method) : undefined,
            'headers': self.headers,
            'body': self.body,
        };
    }
}

File: z-clients/rust/src/models/http_sink_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::http_method::HttpMethod;

/// Configuration for an HTTP sink. The `url`, `headers`, and `body` are templates rendered
/// per-message (see [`diom_core::template_str`]).
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct HttpSinkConfig {
    /// Destination URL.
    pub url: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub method: Option<HttpMethod>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub headers: Option<std::collections::HashMap<String, String>>,

    /// Templated request body. When absent, the raw message value bytes are sent unchanged.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub body: Option<String>,
}

impl HttpSinkConfig {
    pub fn new(url: String) -> Self {
        Self {
            url,
            method: None,
            headers: None,
            body: None,
        }
    }

    pub fn with_method(mut self, value: impl Into<Option<HttpMethod>>) -> Self {
        self.method = value.into();
        self
    }

    pub fn with_headers(
        mut self,
        value: impl Into<Option<std::collections::HashMap<String, String>>>,
    ) -> Self {
        self.headers = value.into();
        self
    }

    pub fn with_body(mut self, value: impl Into<Option<String>>) -> Self {
        self.body = value.into();
        self
    }
}
